secretsharing/ss.go

```diff
@@ -0,0 +1,172 @@
+// Package secretsharing provides methods to split secrets in shares.
+//
+// A (t,n) secret sharing allows to split a secret into n shares, such that the
+// secret can be only recovered given more than t shares.
+//
+// The New function creates a Shamir secret sharing [1], which relies on
+// Lagrange polynomial interpolation.
+//
+// The NewVerifiable function creates a Feldman secret sharing [2], which
+// extends Shamir's by allowing to verify that a share corresponds to the
+// secret.
```

Comment:
I think we should try to define "verifiability" a bit more precisely. What does it mean for a share to "correspond" to the secret?

Reply:
updated text without entering in too much details.